sys-utils/nsenter.c

```diff
@@ -553,15 +537,6 @@ int main(int argc, char *argv[])
 	if (do_user_parent)
 		set_parent_user_ns_fd();
 
-	/*
-	 * Update namespaces variable to contain all requested namespaces
-	 */
-	for (nsfile = namespace_files; nsfile->nstype; nsfile++) {
-		if (nsfile->fd < 0)
-			continue;
-		namespaces |= nsfile->nstype;
-	}
-
```

Comment:
I'm not sure about this. For example, set_parent_user_ns_fd() modifies nsfile->fd, so `namespaces` has to be updated. IMHO, this for() makes code robust as not all is done in the getopt_long() block.

Reply:
TBH, I would like to remove `--user-parent`, I don't know if it's allowed by util-linux backwards compatibility policy. It doesn't work well for the original stated use case (bwrap) because you need to do two nsenter invocations `nsenter --user-parent -t $pid nsenter -U -m -t $pid`, it cannot support more complicated nesting structures (`--user-parent` can only be specified once), and it's unnecessary when using pidfds. The only case it could be useful is for kernels between 4.9 and 5.8 which have `NS_GET_PARENT` but not `pidfd_open`. I can put this loop back though, I don't feel strongly about it.

Reply:
We care about backward compatibility really seriously, so removing an option is an unwanted step.